We need to treat __WINESRC__ same as __MINGW32__, otherwise wine will fail to
build with icu-enabled libxml2; see http://bugs.gentoo.org/show_bug.cgi?id=354745
and http://bugs.gentoo.org/show_bug.cgi?id=355515

diff -ru icu-orig/source/common/unicode/pwin32.h icu/source/common/unicode/pwin32.h
--- icu-orig/source/common/unicode/pwin32.h	2010-02-04 15:32:21.000000000 -0500
+++ icu/source/common/unicode/pwin32.h	2011-02-19 12:39:41.488301169 -0500
@@ -40,7 +40,7 @@
 
 /** Define whether inttypes.h is available */
 #ifndef U_HAVE_INTTYPES_H
-#   if defined(__BORLANDC__) || defined(__MINGW32__)
+#   if defined(__BORLANDC__) || defined(__MINGW32__) || defined(__WINESRC__)
 #       define U_HAVE_INTTYPES_H 1
 #   else
 #       define U_HAVE_INTTYPES_H 0
diff -ru icu-orig/source/common/unicode/umachine.h icu/source/common/unicode/umachine.h
--- icu-orig/source/common/unicode/umachine.h	2010-06-03 11:51:19.000000000 -0400
+++ icu/source/common/unicode/umachine.h	2011-02-19 12:52:17.134931969 -0500
@@ -43,7 +43,7 @@
 
 #if defined(U_PALMOS)
 #   include "unicode/ppalmos.h"
-#elif !defined(__MINGW32__) && (defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64))
+#elif !(defined(__MINGW32__) || defined(__WINESRC__)) && (defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64))
 #ifdef CYGWINMSVC
 #   include "unicode/platform.h"
 #endif
